frontend/src/pages/auth/signup/SignUp.tsx

```diff
@@ -1,12 +1,156 @@
+import PasswordField from "../components/fields/password/PasswordField";
+import {useAppDispatch} from "../../../stores/redux/hooks";
+import {createNotification} from "../../../components/notification/Notification";
+import {Notifications} from "../../../common/constansts/notifications";
+import {showNotification} from "../../../stores/redux/notification/notificationSlice";
+import {FieldsValidation} from "./types";
+import {UserCredentials, UserType} from "../../../types/user";
+import {NotificationType} from "../../../types/notification";
+
+const initialUserCredentials: UserCredentials = {
+    email: "",
+    name: "",
+    password: ""
+}
+
+const initialFieldsValidation: FieldsValidation = {
+    isEmailValid: false,
+    isPasswordValid: false,
+    isNameValid: false
+};
 
 const SignUp: React.FC = () => {
+    const [userCredentials, setUserCredentials] = useState<UserCredentials>(initialUserCredentials);
+    const [selectedRole, setSelectedRole] = useState<UserType>(UserType.Customer);
+    const [agreeChecked, setAgreeChecked] = useState(false);
+    const [fieldsValidation, setFieldsValidation] = useState<FieldsValidation>(initialFieldsValidation);
+    const [isValid, setIsValid] = useState(false);
+    const navigate = useNavigate();
+    const dispatch = useAppDispatch();
+
+    const handleRoleChange = (event: React.ChangeEvent<HTMLInputElement>) => setSelectedRole(event.target.value as UserType);
+    const handleAgreeCheckboxChange = (event: React.ChangeEvent<HTMLInputElement>) => setAgreeChecked(event.target.checked);
+
+    const handleSignUp = async () => {
+        if (isValid) {
+            await signUpService(userCredentials, selectedRole);
+            navigate(Paths.SignIn);
+            const notification = createNotification(Notifications.SIGN_UP, NotificationType.SUCCESS);
+            dispatch(showNotification(notification));
+        } else {
+            const notification = createNotification(Notifications.VALIDATION_NOT_PASSED, NotificationType.ERROR);
+            dispatch(showNotification(notification));
+        }
+    };
+
+    const updateEmailChange = (email: string, isEmailValid: boolean) => {
+        setUserCredentials({...userCredentials, email});
+        setFieldsValidation({...fieldsValidation, isEmailValid});
+    };
+
+    const updateNameChange = (name: string, isNameValid: boolean) => {
+        setUserCredentials({...userCredentials, name});
+        setFieldsValidation({...fieldsValidation, isNameValid});
+    };
+
+    const updatePasswordChange = (password: string, isPasswordValid: boolean) => {
+        setUserCredentials({...userCredentials, password});
+        setFieldsValidation({...fieldsValidation, isPasswordValid});
+    };
+
+    useEffect(() => {
+        setIsValid(
+            fieldsValidation.isEmailValid &&
+            fieldsValidation.isPasswordValid &&
+            fieldsValidation.isNameValid &&
+            selectedRole &&
+            agreeChecked
+        );
```

Comment:
useEffect should not have logic. The only task is to call relevant methods when some deps are changed . Otherwise during the time it can be very big and hard to read